### ISSUE TITLE
Add support for solc `0.8.20`

### ISSUE THREAD

**Issue by pcaversaccio** [foundry-rs/foundry#4916]
solc `0.8.20` just got released https://github.com/ethereum/solidity/releases/tag/v0.8.20

**Comment by mattsse** [comment]
fyi @roynalnaruto

**Comment by Evalir** [comment]
sick—we can get to this as soon as ethers gets a bump with 0.8.20 support. Ideally it would set Shanghai as the default EVM version too, so we can move over everything to it as well to be in line with the rest of the ecosystem.

**Comment by pcaversaccio** [comment]
FYI (for the ones interested), I also added solc `0.8.20` support for dapptools here https://github.com/dapphub/dapptools/pull/984.

**Comment by DaniPopes** [comment]
Set default to "Shanghai" and added support for the `PUSH0` opcode: gakonst/ethers-rs#2414

**Comment by Evalir** [comment]
this was done in https://github.com/foundry-rs/foundry/pull/4975 — should we close this and track the actual upgrade to Shanghai as default as a separate issue or edit this one?

**Comment by pcaversaccio** [comment]
Let's stick to this issue if everyone agrees, so it's easy to find for anyone who searches for the upgrade and default setting to Shanghai. Will update the title my issue accordingly.

**Comment by Evalir** [comment]
Hey @d1onys1us — great questions. This means N1. N2 is actually being done right now on this PR: #5001.

The current default EVM Version is Paris, but will be set to shanghai after ethers-rs is bumped with it as default.

**Comment by pcaversaccio** [comment]
what's the status here?

**Comment by d1onys1us** [comment]
> what's the status here?

adding to this--how do I know if anvil is running --shanghai by default? when i do `cast block` against either `anvil` or `anvil --hardfork shanghai`, I am not seeing `withdrawalsRoot` returned in the block data.

**Comment by AmazingAng** [comment]
How to set the test environment of Foundry to Shanghai Upgrade?

**Comment by pcaversaccio** [comment]
You can do this in the `foundry.toml` via the `evm_version` option; example https://github.com/pcaversaccio/snekmate/blob/main/foundry.toml#L16.

**Comment by AmazingAng** [comment]
> You can do this in the `foundry.toml` via the `evm_version` option; example https://github.com/pcaversaccio/snekmate/blob/main/foundry.toml#L16.

Thanks, I solved it using --evm-version shanghai in cmd.
But we not set default to shanghai?
The Paris version does not support `PUSH0` opcode, which is not good.

**Comment by pcaversaccio** [comment]
> Thanks, I solved it using --evm-version shanghai in cmd.
> But we not set default to shanghai?
> The Paris version does not support `PUSH0` opcode, which is not good.

No, too many EVM-based chains do currently not support `PUSH0`, which would lead to a deployment or even runtime revert if compiled via `shanghai`. Until the to 10 chains in terms of TVL don't support `shanghai`, it's a good choice to default to `paris`.
